BlueprintUICommonControls/Sources/Internal/KeyboardObserver.swift

```diff
@@ -0,0 +1,172 @@
+//
+//  KeyboardObserver.swift
+//  BlueprintUICommonControls
+//
+//  Created by Kyle Van Essen on 2/16/20.
+//
+
+import UIKit
+
+
+protocol KeyboardObserverDelegate : AnyObject
+{
+    func keyboardFrameWillChange(observer : KeyboardObserver)
+}
+
+/**
+ Implementation borrowed from Listable:
+ https://github.com/kyleve/Listable/blob/master/Listable/Sources/Internal/KeyboardObserver.swift
+ 
+ iOS Docs for keyboard management:
+ https://developer.apple.com/library/archive/documentation/StringsTextFonts/Conceptual/TextAndWebiPhoneOS/KeyboardManagement/KeyboardManagement.html
+ */
+final class KeyboardObserver
+{
+    private let center : NotificationCenter
+    
+    weak var delegate : KeyboardObserverDelegate?
+    
+    //
+    // MARK: Initialization
+    //
+    
+    init(center : NotificationCenter = .default)
+    {
+        self.center = center
+
+        self.center.addObserver(self, selector: #selector(keyboardFrameChanged(_:)), name: UIWindow.keyboardWillChangeFrameNotification, object: nil)
+        self.center.addObserver(self, selector: #selector(keyboardFrameChanged(_:)), name: UIWindow.keyboardDidChangeFrameNotification, object: nil)
```

Comment:
Those should probably trigger different behaviors if we want animations to work correctly...

Reply:
Now to remember why I did this... I think `will` is the one we want to use for animations, I forget why I included did..

Reply:
Dug into this more: When there's a floating keyboard or split keyboard, will is not called for frame changes, only did is called once the keyboard position changes. Because we guard based on the last reported frame, animations still work as expected. Adding a comment inline.